WordPress/gutenberg
# Fix the editor save keyboard shortcut not working in code editor view

## Description
Something I spotted in my travels around Gutenberg—saving using the shortcut (Cmd+S/Ctrl+S) only works in the Visual Editor view, when switching to the CodeEditor view it ceases to work.

This PR extracts the save keyboard shortcut into a component `EditorSaveKeyboardShortcut`. That shortcut is now implemented in `TextEditorGlobalKeyboardShortcuts` (a new component) and `VisualEditorGlobalKeyboardShortcuts` (which has replaced `EditorGlobalKeyboardShortcuts`).

The old `EditorGlobalKeyboardShortcuts` component has been deprecated.

## How has this been tested?
1. Create a new post
2. Add a title and some content
3. Use the save shortcut (Cmd+S on a Mac, Ctrl + S elsewhere).
4. Observe that the post is saved
5. Switch to the Code Editor
6. Make some changes to the content
7. Use the save shortcut again
8. Observe that the post is saved (works in this branch, fails in master).




I tested the deprecation by temporarily replacing `VisualEditorGlobalKeyboardShortcuts` with `EditorGlobalKeyboardShortcuts` — the functionality should continue to work and the deprecation message should be displayed in the console.

## Screenshots 

## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
